src/client/pydaos/pydaos_shim.c

```diff
@@ -0,0 +1,249 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#ifdef __USE_PYTHON3__
+#include <python3.6m/Python.h>
+
+/* Those are gone from python3, replaced with new functions */
+#define PyInt_FromLong		PyLong_FromLong
+#define PyInt_AsLong		PyLong_AsLong
+#define PyString_FromString	PyUnicode_FromString
+#define PyString_AsString	PyBytes_AsString
+#else
+#include <Python.h>
+#endif
+
+#include "daos_types.h"
+#include "daos.h"
+#include "daos_obj_class.h"
+#include <daos/object.h>
+
+#define PY_SHIM_MAGIC_NUMBER 0x7A89
+
+static int
+__is_magic_valid(int input)
+{
+	if (input != PY_SHIM_MAGIC_NUMBER) {
+		D_ERROR("MAGIC number doesnt match, expected %d got %d\n",
+			PY_SHIM_MAGIC_NUMBER, input);
+		return 0;
+	}
+
+	return 1;
+}
+
+#if 0 /* avoid flow control in MACRO in order to pass CI */
+/** Macro that parses out magic value and verifies it */
+#define RETURN_NULL_IF_BAD_MAGIC(args)					\
+do {									\
+	int magic;							\
+	if (!PyArg_ParseTuple(args, "i", &magic)) {			\
+		D_ERROR("Bad arguments passed to %s\n", __func__);	\
+		return NULL;						\
+	}								\
+									\
+	if (!__is_magic_valid(magic)) {					\
+		return NULL;						\
+	}								\
+} while (0)
+
+
+/** Parse arguments and magic number out*/
+#define RETURN_NULL_IF_FAILED_TO_PARSE(args, format, x...)		\
+do {									\
+	int magic;							\
+	if (!PyArg_ParseTuple(args, "i"format, &magic, x)) {		\
+		D_ERROR("Bad arguments passed to %s\n", __func__);	\
+		return NULL;						\
+	}								\
+									\
+	if (!__is_magic_valid(magic)) {					\
+		return NULL;						\
+	}								\
+ } while (0)
+#endif
+
+/**
+ * Implementations of shim functions
+ */
+
+static PyObject *
+__shim_handle__test(PyObject *self, PyObject *args)
+{
+	long ret;
+	int magic;
+                                                      \
```

Comment:
(style)  code indent should use tabs where possible
(style)  Avoid unnecessary line continuations